TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/CaptureProxy.java

```diff
@@ -375,4 +420,35 @@ public static void main(String[] args) throws InterruptedException, IOException
         // work in the background.
         proxy.waitForClose();
     }
+
+    static ProxyChannelInitializer buildProxyChannelInitializer(RootCaptureContext rootContext,
+                                                                BacksideConnectionPool backsideConnectionPool,
+                                                                Supplier<SSLEngine> sslEngineSupplier,
+                                                                @NonNull RequestCapturePredicate headerCapturePredicate,
+                                                                List<String> headerOverrides,
+                                                                IConnectionCaptureFactory connectionFactory)
+    {
+        var headers = convertPairListToMap(headerOverrides);
+        return new ProxyChannelInitializer(
+            rootContext,
+            backsideConnectionPool,
+            sslEngineSupplier,
+            connectionFactory,
+            headerCapturePredicate
+        ) {
+            @Override
+            protected void initChannel(@NonNull SocketChannel ch) throws IOException {
+                super.initChannel(ch);
+                for (var kvp : headers.entrySet()) {
```

Comment:
Is this the right place for this logic?

Reply:
What do you mean?

Reply:
It feels weird to have this logic inside CaptureProxy.java, i'd rather keep more logic outside of that class. For the sake of this PR i'm fine with it being in here for now.